### PR TITLE
Update plugin generator to new plugin infrastructure

### PR DESCRIPTION
**Pull request by daffl**
Update the plugin generator to the new infrastructure as proposed in https://github.com/feathersjs/feathers/issues/608

Client side build option will be added in a separate PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "yeoman-test": "^1.4.0"
   },
   "engines": {
-    "node": ">= 5.0.0"
+    "node": ">= 6.0.0"
```
Comment (ekryski) [inline package.json]: You probably don't want to ignore the `lib/` folder. https://github.com/feathersjs/generator-feathers-plugin/blob/12f0db84a89f30d1a00c09e0b0664ae3a9172e3a/package.json#L36

Comment (daffl) [inline package.json]: Shouldn't be ignored anymore no (https://github.com/feathersjs/generator-feathers-plugin/blob/12f0db84a89f30d1a00c09e0b0664ae3a9172e3a/app/templates/package.json#L39)?

**Comment by marshallswain** [comment]
Can this branch be removed?

**Comment by daffl** [comment]
Yes. Done.

**Pull request merged**